src/main/kotlin/org/rust/ide/actions/RustFmtFileAction.kt

```diff
@@ -0,0 +1,31 @@
+package org.rust.ide.actions
+
+import com.intellij.execution.ExecutionException
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.externalSystem.util.ExternalSystemUiUtil
+import com.intellij.openapi.module.ModuleUtilCore
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.ui.MessageType
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.util.PathUtil
+import org.rust.cargo.toolchain.RustToolchain
+
+class RustFmtFileAction : DumbAwareAction() {
+    override fun actionPerformed(e: AnActionEvent?) {
+        val project = e?.project ?: return
+        val file = e?.getRequiredData(CommonDataKeys.VIRTUAL_FILE) ?: return
```

Comment:
`getRequiredData` may throw an exception if the data is not available (for example, the action was invoked outside of the editor). More info in the docstring for this method.